### ISSUE TITLE
resolveStylesheet does not resolve package "exports" subpath patterns (e.g. "./*.css"), so the theme fails to build and no-unknown-classes silently degrades

### ISSUE THREAD

**Issue by camdressler** [shadcn-ui/lint#9]
### What happens

A CSS entry that imports a stylesheet from a dependency whose `package.json` exposes it through an `exports` subpath *pattern* (a key containing `*`) does not resolve. The Tailwind worker throws, the theme is never built, and the plugin prints one line on stderr:

```text
[@shadcn/lint] The Tailwind theme at src/app.css could not be built (@import "demo-widgets/styles.css" could not be resolved from /tmp/repro/src); no-unknown-classes is using the grammar bundled with @shadcn/lint there until it can.
```

Exit code stays 0. From then on `no-unknown-classes` judges every class against the bundled grammar, which knows nothing about the project `@theme`, so every utility the project declares is reported as unknown. The finding count scales with the size of the tree and all of it disappears the moment the `@import` resolves, so the failure reads as a rule problem rather than a resolution problem.

### Minimal reproduction

A fake dependency that ships its CSS behind a pattern export:

```jsonc
// node_modules/demo-widgets/package.json
{ "name": "demo-widgets", "version": "1.0.0", "exports": { "./*.css": "./dist/*.css" } }
```

```css
/* node_modules/demo-widgets/dist/styles.css */
.demo-widget { display: block; }
```

```css
/* src/app.css, the tailwind.css entry in components.json */
@import "tailwindcss";
@import "demo-widgets/styles.css";

@theme { --spacing-card: 1.25rem; }
```

```tsx
// src/page.tsx
export function Page() { return <div className="p-card">Card</div>; }
```

```jsonc
// .oxlintrc.json
{ "jsPlugins": ["@shadcn/lint"], "rules": { "shadcn/no-unknown-classes": "warn" } }
```

Then `npx oxlint src`.

**Expected:** `demo-widgets/styles.css` resolves to `node_modules/demo-widgets/dist/styles.css`, the theme builds, `p-card` is accepted.

**Actual:** the stderr line above, exit code 0, and `p-card` reported as unknown. Deleting the second `@import` makes the report go away, which is the only hint that the import was the cause.

### Cause

`resolveStylesheet` in `dist/tailwind-worker.js` looks a subpath up by exact key only:

```js
const target = exports && typeof exports === "object" ? styleTarget(exports[`./${subpath}`]) : null;
```

Node resolves `exports` keys containing a single `*` by substituting the matched segment into the target, so `"./*.css": "./dist/*.css"` is a valid way to publish `demo-widgets/styles.css`. The exact lookup misses every such key, and the fallback (`stylesheetAt(pkgDir + subpath)`) looks for `node_modules/demo-widgets/styles.css`, which does not exist because the file lives under `dist/`.

### Patch

This resolves the repro. Pattern keys are tried only when the exact key misses, so existing behaviour is unchanged:

```diff
--- a/dist/tailwind-worker.js
+++ b/dist/tailwind-worker.js
@@ -73,7 +73,18 @@ function resolveStylesheet(base, id) {
 	} catch {}
 	const exports = pkg.exports;
 	if (subpath) {
-		const target = exports && typeof exports === "object" ? styleTarget(exports[`./${subpath}`]) : null;
+		let target = exports && typeof exports === "object" ? styleTarget(exports[`./${subpath}`]) : null;
+		if (!target && exports && typeof exports === "object") {
+			for (const key of Object.keys(exports)) {
+				const star = key.indexOf("*");
+				if (star < 0) continue;
+				const pre = key.slice(2, star), post = key.slice(star + 1), sp = subpath;
+				if (sp.startsWith(pre) && sp.endsWith(post) && sp.length >= pre.length + post.length) {
+					const t = styleTarget(exports[key]);
+					if (t) { target = t.replace("*", sp.slice(pre.length, sp.length - post.length)); break; }
+				}
+			}
+		}
 		if (target) return existingFile(path.join(pkgDir, target));
 		return stylesheetAt(path.dirname(path.join(pkgDir, subpath)), path.basename(subpath));
 	}
```

### Separate suggestion: the degraded mode should be loud

A theme that will not build is not a warning, because the rule keeps reporting against a grammar that cannot be right for the project. Either make the build failure a hard error, or have `no-unknown-classes` report nothing for a file whose theme failed to build. #6 is a different cause of the same degraded state, which is part of the argument for making the state itself loud.

### Versions

`@shadcn/lint` 0.1.0, Oxlint 1.83.0, Tailwind CSS v4.

**Comment by leandromatos** [comment]
Same failure surface, third cause: under pnpm the imported package is a symlink into `node_modules/.pnpm/`, and `loadStylesheet` uses the linked path as the base for that file's own imports, so a dependency installed beside the real file is not found and the theme fails to build the same way. Details, a fixture in the pnpm shape and a fix on a branch in #15.
